flink-connectors/flink-hbase/src/main/java/org/apache/flink/table/descriptors/HBase.java

```diff
@@ -0,0 +1,130 @@
+	}
+
+	/**
+	 * Set the Apache HBase version to be used. Required.
+	 *
+	 * @param version HBase version. E.g., "1.4.3".
+	 */
+	public HBase version(String version) {
+		properties.putString(CONNECTOR_VERSION, version);
+		return this;
+	}
+
+	/**
+	 * Set the HBase table name, Required.
+	 *
+	 * @param tableName Name of HBase table.  E.g., "testNamespace:testTable", "testDefaultTable"
+	 */
+	public HBase tableName(String tableName) {
+		properties.putString(CONNECTOR_TABLE_NAME, tableName);
+		return this;
+	}
+
+	/**
+	 * Set the zookeeper quorum address to connect the HBase cluster. Required.
+	 *
+	 * @param zookeeperQuorum zookeeper quorum address to connect the HBase cluster.
+	 *                        E.g., "localhost:2181,localhost:2182,localhost:2183".
+	 */
+	public HBase zookeeperQuorum(String zookeeperQuorum) {
+		properties.putString(CONNECTOR_ZK_QUORUM, zookeeperQuorum);
+		return this;
+	}
+
+	/**
+	 * Set the zookeeper node parent path of HBase cluster. Default to use "/hbase", Optional.
+	 *
+	 * @param zookeeperNodeParent zookeeper node path of hbase cluster. E.g, "/hbase/example-root-znode".
+	 */
+	public HBase zookeeperNodeParent(String zookeeperNodeParent) {
+		properties.putString(CONNECTOR_ZK_NODE_PARENT, zookeeperNodeParent);
+		return this;
+	}
+
+	/**
+	 * Set threshold when to flush buffered request based on the memory byte size of rows currently added.
+	 * Default to <code>2mb</code>. Optional.
+	 *
+	 * @param maxSize the maximum size (using the syntax of {@link MemorySize}).
+	 */
+	public HBase writeBufferFlushMaxSize(String maxSize) {
+		properties.putMemorySize(CONNECTOR_WRITE_BUFFER_FLUSH_MAX_SIZE, MemorySize.parse(maxSize, MemorySize.MemoryUnit.BYTES));
+		return this;
+	}
+
+	/**
+	 * Set threshold when to flush buffered request based on the number of rows currently added.
+	 * Defaults to not set, i.e. won't flush based on the number of buffered rows. Optional.
+	 *
+	 * @param writeBufferFlushMaxRows number of added rows when begin the request flushing.
+	 */
+	public HBase writeBufferFlushMaxRows(int writeBufferFlushMaxRows) {
+		properties.putInt(CONNECTOR_WRITE_BUFFER_FLUSH_MAX_ROWS, writeBufferFlushMaxRows);
+		return this;
+	}
+
+	/**
+	 * Set an interval when to flushing buffered requesting if the interval passes, in milliseconds.
+	 * Defaults to not set, i.e. won't flush based on flush interval. Optional.
+	 *
+	 * @param interval flush interval. The string should be in format "{length value}{time unit label}"
+	 *                 E.g, "123ms", "1 s", If no time unit label is specified, it will be considered as
+	 *                 milliseconds. For more details about the format, please see
+	 *                 {@link TimeUtils#parseDuration(String)}}.
+	 */
+	public HBase writeBufferFlushInterval(String interval) {
+		Duration duration = TimeUtils.parseDuration(interval);
+		properties.putLong(CONNECTOR_WRITE_BUFFER_FLUSH_INTERVAL, duration.toMillis());
```

Comment:
We can keep the original interval text into the properties to make the property more readable. 
In the future, we want to prohibit numeric if the property accepts a duration string, because numeric is ambiguous.

Reply:
OK